src/prefect/server/schemas/schedules.py

```diff
@@ -324,6 +324,10 @@ def _get_dates_generator(
         # as an event (if it meets the cron criteria)
         start = start.subtract(seconds=1)
 
+        # Respect microseconds by rounding up
+        if start.microsecond > 0:
```

Comment:
this is the main difference from the original pr, that we apply microsecond rounding prior to generating a localized and naive timestamp